src/routes/user/user.service.ts

```diff
@@ -126,3 +128,27 @@ export const verifyEmail = async (token: string, email: string): Promise<any> =>
 
   return Format.badRequest('Invalid verification link');
 };
+
+/**
+ * Refresh Access Token
+ *
+ * @param {user} user - user
+ */
+export const refreshAccessToken = async (user: UserDoc | undefined): Promise<any> => {
+  logger.log(`[${SERVICES_NAMES.refreshAccessToken}] is called`);
+
+  if (!user) {
+    return Format.notFound('User not found');
+  }
+
+  const { accessToken, refreshToken } = generateTokens({ id: user.id, email: user.email });
```

Comment:
We're rotating the refresh token here which is generally not recommended.

**For example,**
At the time of first login we're generating access token as well as refresh token. Now the access token expires every **15 minutes**, so new access token should be generated. But here we're generating new access token as well as new refresh token. Now we end up with 1 **extra** refresh token which we don't need every time our access token expires.

There's a use case of this as well - **remember me feature**. When user don't want to sign in again after 7 days, the refresh token should be rotated regularly to avoid expiry. I think we'll need to create a separate issue to handle that.

For now what you can do is only destructure the accessToken and send accessToken only in the response.

Also if you've got any idea on the optimal way to implement the refresh token rotation, please give your thoughts.